tests/third_party/sqlite/CMakeLists.txt

```diff
@@ -16,3 +16,4 @@ add_custom_command(
     ${CMAKE_COMMAND} -E tar xfz ${SQLITE3_ARCH_NAME})
 
 add_library(sqlite3 sqlite-amalgamation-3210000/sqlite3.c)
+target_link_libraries(sqlite3 dl pthread)
```

Comment:
Since it's actually `sqlite3` that depend on `dl` and `pthread`. So these dependencies have been removed from `example` and `tests` above.